fairroot/base/sim/FairModule.cxx

```diff
@@ -280,12 +280,14 @@ void FairModule::AddSensitiveVolume(TGeoVolume* vol)
     auto volName = vol->GetName();
     LOG(debug2) << "AddSensitiveVolume " << volName;
 
-    auto addedVol = vList->addVolume(std::make_unique<FairVolume>(volName, fNbOfVolumes));
-    if (!addedVol) {
-        return;
+    if (!vList->findObject(volName)) {
```

Comment:
Sry, no. This check was just recently removed because it is already performed by `addVolume()` and thus was performed twice in the past which you now propose to revert to.

`addVolume()` now returns the result of the `findObject()` check so a caller knows whether the volume was added. `vList` effectively resembles the semantics of a [*Set*](https://en.wikipedia.org/wiki/Set_(abstract_data_type)).

Reply:
I fully agree. This only hides the error message.

If you just want to hide the error message, then
1. Just remove it in FairVolumeList::addVolume and make sure that all callers have proper logging (maybe at debug level).
2. Or give addVolume an optional quiet option like this `addVolume(… volume, bool quiet=False)` and call it with `, true)` if you want to calm it.

I'd rather prefer (1) instead of (2).

Reply:
I have changed the commit according to (1):
`LOG(error)` was replaced with `LOG(debug)`